### PR TITLE
Add Map.remove_layer method

### PR DESCRIPTION
**Pull request by zdila**
This fixes https://github.com/mapnik/node-mapnik/issues/467. Tests are missing.

### CODE REVIEW

**Comment by zdila** [comment]
I've added tests and updated changelog. Could it please somebody review and merge?

File: src/mapnik_map.cpp [hunk 1]
```diff
@@ -936,6 +937,25 @@ NAN_METHOD(Map::add_layer) {
...
+ * @param {mapnik.Layer} new layer
+ */
+NAN_METHOD(Map::remove_layer) {
+    if (!info[0]->IsNumber()) {
+        Nan::ThrowTypeError("index must be number");
+        return;
+    }
+
+    Map* m = Nan::ObjectWrap::Unwrap<Map>(info.Holder());
+    m->map_->remove_layer(info[0]->IntegerValue());
```
Comment (artemp) [inline src/mapnik_map.cpp]: What would happen if layer index is outside of valid range e.g `<0` or `>num_layers` ? 'remove_layer' calls `erase(begin+ index)`, I think we need an extra validation logic to check layer index is valid, what do think?

Comment (zdila) [reply src/mapnik_map.cpp]: I've improved arguments validation.

Comment (artemp) [reply src/mapnik_map.cpp]: Great, thanks.

**Pull request merged**